lib/database.h

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: 2021 Tobias Fella <fella@posteo.de>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <QtCore/QObject>
+#include <QtSql/QSqlQuery>
+#include <QtCore/QVector>
+
+#include "crypto/e2ee.h"
+
+namespace Quotient {
+class Database : public QObject
+{
+    Q_OBJECT
+
+public:
+    static Database &instance()
```

Comment:
I feel a bit uneasy that a `Database` instance can, at least theoretically, be produced after `QApplication` is disassembled, since `QSqlDatabase` relies on a live application object. Not sure how to handle it best yet. Parenting this object to `qApp` is not good enough, since children are deleted from `QObject` destructor - by then, `QApplication` interface no more exists.

Reply:
i guess we can check in the constructor if there is a qcoreapplication instance, would that be enough?

Reply:
Does it sound too far-fetched if `Connection` owns `Database`? Guess we want to keep separate accounts in separate databases, right? (That also means that the current path to the database should be adjusted.)

Reply:
Done